packages/sanity/src/core/studio/copyPaste/useCopyPasteAction.ts

```diff
@@ -0,0 +1,333 @@
+        })
+        return
+      }
+
+      const fields = payloadValue.items.map((item) => item.schemaTypeTitle).join('", "')
+
+      toast.push({
+        status: 'success',
+        title: isDocument
+          ? t('copy-paste.on-copy.validation.copy-document-success.title', {
+              fieldNames: fields,
+            })
+          : t('copy-paste.on-copy.validation.copy-field_one-success.title', {
+              fieldName: fields,
+            }),
+      })
+    },
+    [documentMeta, setCopyResult, telemetry, toast, t],
+  )
+
+  const onPaste = useCallback(
+    async (targetPath: Path, value: FormDocumentValue | undefined, options?: PasteOptions) => {
+      const {schemaType: targetDocumentSchemaType} = documentMeta || {}
+      const targetSchemaType = resolveSchemaTypeForPath(
+        targetDocumentSchemaType!,
+        targetPath,
+        value,
+      )!
+
+      const clipboardItem = await getClipboardItem()
+
+      // Return early if no clipboard item or if clipboard item is invalid
+      if (!clipboardItem) {
+        toast.push({
+          status: 'info',
+          title: t('copy-paste.on-paste.validation.clipboard-empty.title'),
+        })
+        return
+      }
+
+      if (!clipboardItem.documentType) {
+        toast.push({
+          status: 'error',
+          title: t('copy-paste.on-paste.validation.clipboard-invalid.title'),
+        })
+        return
+      }
+
+      const sourceDocumentSchemaType = schema.get(clipboardItem.documentType)
+
+      if (!sourceDocumentSchemaType) {
+        toast.push({
+          status: 'error',
+          title: t('copy-paste.on-paste.validation.clipboard-invalid.title'),
+        })
+        return
+      }
+
+      const updateItems: {patches: FormPatch[]; targetSchemaTypeTitle: string}[] = []
+      const copiedJsonTypes: string[] = []
+
+      for (const item of clipboardItem.items) {
+        const sourceSchemaType = resolveSchemaTypeForPath(
+          sourceDocumentSchemaType,
+          item.documentPath,
+          value,
+        )
+
+        if (!sourceSchemaType) {
+          toast.push({
+            status: 'error',
+            title: t('copy-paste.on-paste.validation.schema-type-incompatible.title', {
+              path: PathUtils.toString(item.documentPath),
+            }),
+          })
+          return
+        }
+
+        if (!targetDocumentSchemaType) {
+          toast.push({
+            status: 'error',
+            title: t('copy-paste.on-paste.validation.schema-type-incompatible.title', {
+              path: PathUtils.toString(targetPath),
+            }),
+          })
+          return
+        }
+
+        const targetSchemaTypeTitle = targetSchemaType.title || targetSchemaType.name
+        const transferValueOptions = {
+          sourceRootSchemaType: sourceSchemaType,
+          sourcePath: [],
+          sourceValue: item.value,
+          targetRootSchemaType: targetSchemaType,
+          targetPath: [],
```

Comment:
why are these hardcoded to empty array? are they being used or can they be removed?